### PR TITLE
[MIG][12.0] fieldservice_substatus

### PR DESCRIPTION
**Pull request by murtuzasaleh**
Field Service - Sub-Status
====================

### CODE REVIEW

**Comment by murtuzasaleh** [comment]
@osi-scampbell @max3903 Can you please review this PR.

File: fieldservice_substatus/models/fsm_order.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        default=lambda self: self._default_stage_id().sub_stage_id)
+
+    @api.multi
+    def write(self, vals):
+        if 'stage_id' in vals:
+            sub_stage_id = self.env['fsm.stage'].browse(
+                vals.get('stage_id')).sub_stage_id
+            if sub_stage_id:
+                vals.update({'sub_stage_id': sub_stage_id.id})
+        return super().write(vals)
```
Comment (brian10048) [inline fieldservice_substatus/models/fsm_order.py]: `return super(FSMOrder, self).write(vals)`

File: fieldservice_substatus/models/fsm_stage_status.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    def _search(self, args, offset=0, limit=None, order=None, count=False,
+                access_rights_uid=None):
+        context = self._context or {}
+        if context.get('fsm_order_stage_id'):
+            stage_id = self.env['fsm.stage'].browse(context.get(
+                'fsm_order_stage_id'))
+            sub_stage_ids = stage_id.sub_stage_id + stage_id.sub_stage_ids
+            if sub_stage_ids:
+                args = [('id', 'in', sub_stage_ids.ids)]
+        return super()._search(args, offset, limit, order, count=count,
```
Comment (brian10048) [inline fieldservice_substatus/models/fsm_stage_status.py]: `return super(FSMStageStatus, self)._search(args, offset, limit, order, count=count,`

File: fieldservice_substatus/models/fsm_order.py (hunk 1 above)
Comment (brian10048) [reply fieldservice_substatus/models/fsm_order.py]: I noticed these minor errors in v11 code when using super

**Review by osi-scampbell — APPROVED** [review]

**Review by max3903 — APPROVED** [review]

**Pull request merged**